tests/bdd/features/volume/topology/test_node_topology.py

```diff
@@ -0,0 +1,519 @@
+    """Return the pool names whose node has the given labels."""
+    node_with_node_labels = get_node_names_with_its_corresponding_node_labels()
+    qualified_nodes = list()
+    if "=" in node_label:
+        # Splitting the pool_label into parts
+        split_parts = node_label.split('=')
+        # Accessing the parts after splitting
+        node_key = split_parts[0]
+        node_value = split_parts[1]
+        for pool_id, pool_node_labels in node_with_node_labels.items():
+            for key, value in pool_node_labels.items():
+                if key == node_key and value == node_value:
+                    qualified_nodes.append(pool_id)
+    else:
+        node_key = node_label
+        node_value = ""
+        for pool_id, pool_labels in node_with_node_labels.items():
+            for key, value in pool_labels.items():
+                if key == node_key:
+                    qualified_nodes.append(pool_id)
+    return qualified_nodes
+
+
+# Return the create volume request body based on the input parameters.
+def create_volume_body(replica, volume_node_topology_inclusion_label):
+    """Create a volume body."""
+    if volume_node_topology_inclusion_label == "alpha1=beta1":
+        topology = Topology(
+            node_topology=NodeTopology(
+                labelled=LabelledTopology(
+                    exclusion={},
+                    inclusion={"alpha1": "beta1"},
+                )
+            )
+        )
+    elif volume_node_topology_inclusion_label == "alpha2=beta2":
+        topology = Topology(
+            node_topology=NodeTopology(
+                labelled=LabelledTopology(
+                    exclusion={},
+                    inclusion={"alpha2": "beta2"},
+                )
+            )
+        )
+    elif volume_node_topology_inclusion_label == "alpha3=beta3":
+        topology = Topology(
+            node_topology=NodeTopology(
+                labelled=LabelledTopology(
+                    exclusion={},
+                    inclusion={"alpha3": "beta3"},
+                )
+            )
+        )
+    elif volume_node_topology_inclusion_label == "alpha1":
+        topology = Topology(
+            node_topology=NodeTopology(
+                labelled=LabelledTopology(
+                    exclusion={},
+                    inclusion={"alpha1": ""},
+                )
+            )
+        )
+    elif volume_node_topology_inclusion_label == "alpha2":
+        topology = Topology(
+            node_topology=NodeTopology(
+                labelled=LabelledTopology(
+                    exclusion={},
+                    inclusion={"alpha2": ""},
+                )
+            )
+        )
+    elif volume_node_topology_inclusion_label == "alpha3":  
+        topology = Topology(
+            node_topology=NodeTopology(
+                labelled=LabelledTopology(
+                    exclusion={},
+                    inclusion={"alpha3": ""},
+                )
+            )
+        )
```

Comment:
hmm we can do this only once?

```suggestion
key, _, value = volume_node_topology_inclusion_label.partition("=")
 topology = Topology(
            node_topology=NodeTopology(
                labelled=LabelledTopology(
                    exclusion={},
                    inclusion={key.strip(): value.strip()},
                )
            )
        )
```